titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/GlobalAgentClusterConstraint.java

```diff
@@ -75,31 +71,23 @@ public void prepare() {
 
     @Override
     public Result evaluate(TaskRequest taskRequest, VirtualMachineCurrentState targetVM, TaskTrackerState taskTrackerState) {
-        Pair<Boolean, String> health = evaluateHealthy(targetVM);
-        if (!health.getLeft()) {
-            return new Result(false, health.getRight());
+        if (!isHealthy(targetVM)) {
+            return UNHEALTHY;
```

Comment:
one downside of this is that we lose some context on why the agent was unhealthy. Was that even useful and exposed somewhere? (e.g.: `/api/v2/scheduler/failures/:id`)